### ISSUE TITLE
Error after connecting to Tor Node

### ISSUE THREAD

**Issue by BenGWeeks** [ZeusLN/zeus#957]
### Describe the bug

After scanning the QR node in Umbrel to a Tor node, the following error was received:

![image](https://user-images.githubusercontent.com/5950171/164744204-742c368c-6410-4543-9d96-433616fbd9b3.png)

An application restart was required.

I would propose that:

1. The error is handled and caught, with a user-friendly message supplied (that helps the user know why, and what to do).
2. The application allows more time to connect to the Tor node.

### Reproduce

To reproduce the error:

1. Install Zeus
2. Connect to your node (Umbrel, Tor)

### Zeus version

v0.6.2

### Network

Tor

### Device

iPhone 13 Pro Max

### Device operating system

iOS 15.4.1

**Comment by Fjlehner** [comment]
I’m writing to report the same issue and also propose the dev team consider a more user-friendly error message at a minimum. 


![A37EC544-6D60-4AD2-A12D-DDFF43DCE3CA](https://user-images.githubusercontent.com/104231547/164792425-a2128847-e777-43f4-8ec6-654e086b5579.png)

**Comment by Mexxblitz** [comment]
Is there any solution to this Problem? I experience the same bug and restarting does not help.

**Comment by Bosch-0** [comment]
Does it work over clearnet? Will get on to a more friendly error message in the upcoming release.

Are you connecting over REST, not gRPC (not supported currently)

**Comment by cmdruid** [comment]
This isn't really a bug IMO. Zeus is being rejected by your Tor node, with the response provided by the Tor node.

Maybe this can be a feature request. I like to see the network error received by Zeus though, I wouldn't want that hidden or obfuscated from me.

**Comment by Bosch-0** [comment]
We'll have a better, non Tor, still private, way for LND users to connect in the future. Tor is a big issue for many users

**Comment by Fjlehner** [comment]
> This isn't really a bug IMO. Zeus is being rejected by your Tor node, with the response provided by the Tor node.
> 
> 
> 
> Maybe this can be a feature request. I like to see the network error received by Zeus though, I wouldn't want that hidden or obfuscated from me.

Interesting. I also use Bluewallet and Fully Noded to connect to my node via Tor. Bluewallet is also hit or miss although generally has far less issues than Zeus. Fully Noded seldom has issues. Lately, Zeus hasn't been connecting at all so it's a completely useless app for me at the moment. Multiple app reinstallation attempts and restarts of my node haven't fixed it either. 

Looking forward to this alternate way to connect w/o using Tor.

**Comment by Fjlehner** [comment]
I've found a bit of a workaround if your node runs Umbrel (as mine does) which is to use Tailscale as a workaround to Tor. 



[https://community.getumbrel.com/t/how-to-use-umbrel-with-tailscale/6782](https://community.getumbrel.com/t/how-to-use-umbrel-with-tailscale/6782)

Image to help: 

![image](https://user-images.githubusercontent.com/104231547/167061745-b5f8cbf5-b061-459d-8991-4685bc6fbba6.jpeg)

**Comment by BenGWeeks** [comment]
Yeah, Tor seems to have an issue with many mobile applications (e.g. BlueWallet). I suspect the issue may be at iOS level and currently testing TailScale. However, I'm not keen on the registration requirements (Google/Microsoft, collecting profile information, ability to know devices, etc.)

**Comment by kaloudis** [comment]
@BenGWeeks have you tried Lightning Node Connect yet?

**Comment by BenGWeeks** [comment]
> @BenGWeeks have you tried Lightning Node Connect yet?

Nope, but shall take a look.

(Although, I do think this error should still be handled by the application with a user-friendly UX).

**Comment by vysegrad** [comment]
is there a non-Tor way to remotely connect yet for iOS users?

**Comment by kaloudis** [comment]
@vysegrad Lightning Node Connect or Tailscale

**Comment by kaloudis** [comment]
The Tor library has since been upgraded to https://github.com/niteshbalusu11/react-native-nitro-tor - please feel free to re-open if this issue arises again
